Update to Version 2.0.0

robot256 opened the pull request:
This is the version I released on the Mod Portal as 2.0.0.  It contains my entire script rewrite for Factorio 0.17.  Non-English translations still need to be updated.

robot256 committed:
Tested overlaying cargo plane sprites

Doesn't work!  Because Aircraft mod sprite sheets have 36 rotations, and Vehicle Wagon requires 128 to be able to overlay the sprite on top of the wagon.

robot256 committed:
First attempt at a cargo plane skin for the vehicle wagon.

robot256 committed:
Updated cargo wagon sprites with better rotation and offsets.

robot256 committed:
Added new graphics from YuokiTani for cargo plane!

robot256 committed:
Merge branch 'master' into addgraphics

robot256 committed:
Added YuokiTani's graphics for Jet-On-Wagon

robot256 committed:
Forgot item for loaded-vehicle-wagon-jet

robot256 committed:
Added gunship wagon model

robot256 committed:
Fix gunship typos

robot256 committed:
Rewrite complete!

Changes:
- Added debug log dump (from Dispatcher mod)
- Handle on_entity_died and script_raised_destroy event to purge loaded wagon data.
- Handle on_marked_for_deconstruction event to give a warning when no space to unload vehicle.
- Handle on_robot_pre_mined event to unload wagon before mining.
- Changed wagon replacement to use Robot256Lib replaceCarriage function.
    - Correctly restores coupling state of new wagon.
    - Preserves mark-for-deconstruction so that robot can still mine empty wagon after it is automatically unloaded.
- Used the multi-entry find_filtered_entities feature to condense the code.
- Abstracted loadWagon and unloadWagon functions, which perform the actual vehicle saving and restoring.
- Handled case where stored vehicle contains equipment that no longer exists in the game (unknown equipment is lost).
- On_configuration_change, purge unknown player and wagon data from global tables.
- Fixed process_tick so that if more than one player clicks on exactly the same tick, they will both be processed correctly.
- Handle mining/unloading of loaded wagon which has no stored vehicle data.

Todo:
- Add new error messages to locale file
- Clean up loaded-wagon selection/finding list

robot256 committed:
Whitespace

robot256 committed:
Simplified loaded wagon selection; switched blueprint mapping to Robot256Lib

robot256 committed:
More cleanup

tab->space
return->elseif

robot256 committed:
Move inventory and grid saving to Robot256Lib functions

robot256 committed:
Added inventory migration and burner saving library

- Added Shall's Rocket Artillery to the loaded-tank class.
- Added migration code from old inventory saving to new.
- Moved burner saving to library code

robot256 committed:
Code cleanup

- Moved get/insert functions inline.
- Moved migration to separate function.
- Moved blueprint handlers to inline.

robot256 committed:
New graphics for Gunship Wagon

robot256 committed:
Bugfixing

- Fix crash when unloading a vehicle that was removed by mod uninstall.
- Fix that winch item could be lost during use.

robot256 committed:
Bugfix

Prevent duplicates in global.loadedWagonList.

robot256 committed:
Delete control_OLD.lua

robot256 committed:
Bugfixes

Deduplication of wagon tables works properly now.

robot256 committed:
Synchronized vehicle and wagon orientations when loading and unloading.

Tries to match the orientation of vehicle when creating a loaded wagon, and match the rotation of the wagon when creating a vehicle.

Some loaded-wagon sprites are flipped relative to vehicle orientation, so global.loadedWagonFlip is set to true in those cases.

robot256 committed:
Robots deconstructing a loaded wagon carry away all the vehicle's contents and grid.

robot256 committed:
Handle when robots pick up non-existent items.

robot256 committed:
Finished robot mining code

- Robots now obey worker_storage_bonus limit of the robot's force.
- Correctly handles stacks with ammo/durability set when doling it out in robot-sized bites.
- Proper prototype existence checking for equipment and vehicles now.

robot256 committed:
Added handler for loaded-wagon ghosts.

robot256 committed:
Fix error message when mining wagon during loading/unloading.

- Fix typo in ghost handling code.
- Add data purge when wagon currently under request is mined/unloaded early.

robot256 committed:
Added settings to change loaded wagon weight

- When enabled, loaded wagon is based on the weight of the (simplest) vehicle it contains.
- Added compatibility with TrainOverhaul by basing weights on the cargo-wagon after rebalancing.

robot256 committed:
Tweaked wagon weight balance

robot256 committed:
Super Rewrite This Time

Major restructuring of code, global variables, and dependencies.

robot256 committed:
Changed to use new stackList manipulation library functions.

robot256 committed:
Update to use new library functions

- No more error message when robots unload loaded wagon.
- Correctly spills items that don't fit in unloaded vehicle.
- Added Schall's Super Heavy Tanks to the unsupported vehicles list.
- Improved specificity of some vehicle entity selections.
- Fixed crash when mining loaded wagon after its vehicle was uninstalled.
- Fixed that player might not receive vehicle item after mining.
- Fixed some non-local variables.

robot256 committed:
Update control.lua

Just in case we ever get a SimpleItemStack without "count" specified, it should be assumed to be 1.

robot256 committed:
Added handlers for dropped items etc

- When a vehicle wagon is marked for deconstruction, all related player actions are halted.
- When a loaded-vehicle-wagon item is picked up, mined by player, or mined by robot, it is converted to vehicle-wagon.

robot256 committed:
Update control.lua

Added event subscriber for on_picked_up_item.

robot256 committed:
Moved Major Event Handlers to separate files

- Moved OnPlayerUsedCapsule, OnPrePlayerMinedItem, and OnRobotPreMined code to separate files in the script directory.
- Created directory to store migration code.

robot256 committed:
Update control.lua

Fixed blueprintLib now that functions are locally defined.

robot256 committed:
Added beams to indicate loading and unloading

robot256 committed:
Updated Error Messages, Locale, and Robot Setting

- Added functionality to setting that allows robots to unload wagons.
- Updated EN locale and added some new error messages.
- Refined messages player receives when mining and destroying wagons.
- Changed wagon mining so that player/robot is given the mining product item of a particular entity, rather than the item of the same name.  This is for compatibility with Cargo Ships' "Boat", among other things.
- Changed loading/unloading distance from 9 to 10.

robot256 committed:
Changed Winch capsule to stack size of 1 - Finally!

Uses the capsule_action parameter "uses_stack = false" so that the winch item never leaves the player's hand or inventory when used.

robot256 committed:
Docs pulled in from Artanis' 1.2.14 update on Mod Portal

robot256 committed:
Merge branch 'master' into scriptRewrite

robot256 committed:
Added data migration code

robot256 committed:
Finished debugging migration code

robot256 committed:
More user interface tweaks

- Winch applies to any vehicle or wagon your cursor selects.  Now you dont' have to click on the exact center of the wagon or car for it to work.
- Unloading now verifies the user requested position before actually unloading.  Now you can unload two cars on top of each other and they space out properly.
- Mining or destroying a vehicle while it is being loaded is handled immediately instead of when the loading timer expires.
- Truck sprite is now properly dependent on the bigtruck mod being installed.
- Set loading distance back to 9 tiles (capsule range is still 10).

robot256 committed:
More cleanup

- Removed outdated AAI interface and dependencies.
- Fix typo in items.lua.
- Comment typos.

robot256 committed:
Updated changelog and Readme

robot256 committed:
Compressed new graphics with TinyPNG.com

robot256 committed:
Update publish_exclude.txt

robot256 committed:
Update info.json

robot256 committed:
Fix bug in selecting entities

- When clicking with winch, treat non-vehicle, non-vehicle-wagon entities as empty space.

robot256 committed:
changelog prepping for 2.0.1

robot256 committed:
Better fix for clicking-on-trees problem.

ST-DDT commented on Readme.txt:
```diff
@@ -1,58 +1,56 @@
 --------
 
-The flatbed wagon graphics are by the extremely talented and gracious YuokiTani!
+German translation by bNarFProfCrazy (SD-DDT).
```
Just ST-DDT

robot256 committed:
Update Readme.txt

Corrected names

ST-DDT commented on Readme.txt:
```diff
@@ -2,7 +2,7 @@ Vehicle Wagon 2.0.0
 
 Version 2.0.0 was released January 23, 2020, was tested using Factorio v0.17.79, and was authored by robot256 based on the original mod of Supercheese and original graphics from YuokiTani and others.
-Additional contributions from: CrazyMooncat, The_Destroyer, Phasma Felis, Brant Wedel, legendblade, narc, Artanis_Mattias, bNarFProf.
+Additional contributions from: CrazyMooncat, The_Destroyer, Phasma Felis, Brant Wedel, legendblade, narc, Artanis_Mattias, SD-DDT.
```
```suggestion
Additional contributions from: CrazyMooncat, The_Destroyer, Phasma Felis, Brant Wedel, legendblade, narc, Artanis_Mattias, ST-DDT.
```

ST-DDT commented on Readme.txt:
```diff
@@ -42,7 +42,7 @@ Direct compatibility with AAI Programmable Vehicles is untested.  Some very outd
 --------
 
-German translation by bNarFProfCrazy (SD-DDT).
+German translation by SD-DDT.
```
```suggestion
German translation by ST-DDT.
```

robot256 committed:
Update to Factorio 0.18

- Fix crash when unloading at angles just west of north.

robot256 committed:
Release v2.1.0

robot256 committed:
Update Readme.txt

Fix names again.

robot256 committed:
Removed dependencies on StdLib mod

- Replaced String.contains with native string.find.
- Replaced Position.distance with native math functions.

robot256 committed:
Added loading ramp (transport belt) beam

Altered base game transport-belt.png into a symmetrical beam sprite.

ST-DDT committed:
Update German translation

robot256 committed:
Compatibility with Mobile Factory

- Save and restore vehicle "last_user", so that unloaded vehicle is properly configured by Mobile Factory.
- Raise "script_raised_destroy" event when loading vehicle.

robot256 committed:
More compatibility with Mobile Factory and Creative Mod

- Fixed bug in handler for loaded wagon ghosts.
- Added correct handling for Creative Mod Instant Deconstruct invisible robots.
- Fixed last-user restore functionality so Mobile Factory works correctly.

robot256 committed:
Update OnRobotPreMined.lua

Syntax error

robot256 committed:
Update to version 2.18.2

- Additional changes to add the conveyor belt loading ramp beam.
- Fixed bug in compatibility code to stop unnecessary

robot256 committed:
Merge pull request #2 from ST-DDT/translation/de

Update German translation

robot256 committed:
Preparing for 2.18.3 - GCKI compatibility

Added compatibility with Gizmo's Car Keys (Improved) [GCKI] and Autodrive.  Includes a setting to determine whether claimed or locked cars can be loaded by non-owners.

Added localized vehicle name to some error/info messages.

Added header to the Polish and Russian locale files so they are not quite as broken.

robot256 committed:
Locale update

Made name and description of loaded wagons dynamic to use the source mod's vehicle names.

robot256 committed:
Update German translation, fix GCKI bugs

Locale updates thanks to Pi-C:
* New German translation.
* Added name references to locale strings to use the source mod's translations.
* Fixed mod setting maximum/minimum parameters
* Fixed locked/owned error message print.

Still need to decide whether "locked" is a subset of "owned" as far as GCKI is concerned.

robot256 committed:
Bugfixes for previous commit.

robot256 committed:
Change GCKI behavior

Since owner and locker are orthogonal, only one option is used and will enable verification that both of them match the player.  This is primarily an anti-griefing measure.  It does not stop players from mining or destroying the loaded wagon and obtaining the contents that way, yet.

Changed the "tutorial" message to always display when unloading, so you can tell what is inside a loaded wagon.

robot256 committed:
Update OnPlayerUsedCapsule.lua

Fix bug in last commit.

robot256 committed:
Specialization for Schall's Tanks

Added new entities for Schall's Light Tank and Schall's Heavy Tank, with weights to match.

Added call to GCKI interface to see owner of loaded vehicle still owns it when attempting to unload.

robot256 committed:
Finished compatibility with Train Overhaul

- Change wagon speed to match Train Overhaul nuclear locomotive when installed.
- Adjust default settings so it is well balanced with both Train Overhaul and Vanilla.
- Update changelog with changes so far.

robot256 committed:
Settings and interface

Added beta model for Krastorio2 Advanced Tank.
Changed selection visual from "gui_arrow" to rendered circles.
Added interface for GCKI to tell when a player no longer owns a stored vehicle.
Added handling when a player is removed from the game while owning or locking a stored vehicle.
Added setting "maximum-weight" to limit which vehicles are allowed to be loaded.
Cleaned up item generation for non-placeable entities.
Moved capsule and loading range values to config.lua

robot256 committed:
Selection error message improvement.

Added vehicle-in-motion error.
Separated "too-far-away" error into "wagon too far from vehicle" and "location too far from wagon" messages.

robot256 committed:
Selection error message improvement.

Added vehicle-in-motion error.
Separated "too-far-away" error into "wagon too far from vehicle" and "location too far from wagon" messages.

robot256 committed:
Merge branch 'master' of https://github.com/robot256/Vehicle-Wagon-2

robot256 committed:
Update OnPlayerUsedCapsule.lua

robot256 committed:
Changed Unloading range to oblong

Changed unloading range to 6 tiles from the centerline of the wagon for a more realistic area.
Fixed some accidental globals.

robot256 committed:
Improved checking for vehicle movement

on_tick now runs whenever any vehicle is selected.  As soon as the wagon or car moves, it will be deselected.

robot256 committed:
Added weight calculation re-checking

Added weight re-checking to data-final-fixes phase.
Added check for orphaned wagon_data during on_configuration_changed.
Added migration to convert wagon_data[id].last_user from LuaPlayer to player_index, to avoid storing any object references in the table.

robot256 committed:
Fixed GCKI interface. Added other mod migration detection.

Fixed typos in release_owned_by_player.
Removed redundant [item-name] locale keys (they are ignored when a matching [entity-name] is found).
Added vehicle prototype error detection in on_configuration_changed.

robot256 committed:
Added wagon contents alt-mode icons

robot256 committed:
Remove Autodrive owner when player removed from game

robot256 committed:
Added rounded head for loading ramp sprite

Also fixed bug in GCKI code to make wagons unminable when owner/locker is set.

robot256 committed:
Moved loading ramp beam creation to script/renderVisuals.lua

Made loading and unloading ramps two separate entities.
Moved beam creation to helper functions in renderVisuals for later elaboration.
Tweaked loaded contents icon size.
Moved loading effect duration (120 ticks) to config.lua.

robot256 committed:
Updated loading ramp sprites

robot256 committed:
Vastly improved loading ramp sprites and behavior

Loading/unloading ramps now point toward the graphical center of the flatbed and stop when they reach the edge of the bed.
Loading ramps start some distance from the vehicle instaed of on top of it.

robot256 committed:
Improved unloading and ramp behavior

- Changed unloading behavior so that vehicles always point outwards, driving off the ramp.
- Added custom boolean parameter "vehicle_loaded" to "script_raised_destroy" event when vehicle was loaded successfully.
    - Added custom boolean parameter "vehicle_unloaded" to "script_raised_built" event when vehicle was unloaded successfully.
- Added mininum distance based on vehicle size to avoid accidentally destroying the wago when unloading.

robot256 committed:
Added check for GCKI Version

Ignore GCKI if required interface functions don't exist (old version installed).

robot256 committed:
Reorganized graphics/ramp folder

robot256 committed:
Release for 2.18.3

robot256 committed:
Fixes to GCKI compatibility.

robot256 committed:
Fixed last commit

- Fixed last commit so it will run
- Added wagon_data[unit_number].wagon field to store reference to loaded wagon entity.  This is only used to alter the minable property according to GCKI compatibility code.
- Added migration in on_configuration_change to add entity reference to data table.
- Any other time the entity reference is not found, it is ignored.  Wagons stuck like that will need to be unloaded and reloaded before their minable property will be set right.

robot256 committed:
Fix more bugs in last commit

- Fixed typos in migration code.
- Made variable names more uniform.

robot256 committed:
Fix dependency issue. Update for 2.18.5

- Moved all modded loaded wagons to data-updates, so that they will be sure to be created after their respective vehicles.

robot256 committed:
Actually that was 2.18.4

robot256 committed:
Update publish_mod.bat

robot256 committed:
Changes from Pi-C

- Pi-C changed the GCKI and AUtodrive mod interface to "restore vehicle data" via custom parameters on the script_raised_built event.  Now the interface is only used to retrieve the settings for saving.
- Updates to localse.

robot256 committed:
Made GCKI setting always visible and removed dependency

robot256 committed:
First step moving 1.x to 2.x to migration code

robot256 committed:
Save and restore all the entity flags

- Preserve on vehicle load and unload: minable, destructible, operable, rotatable, enable_logistics_while_moving.
- Raise events when replacing wagons.

robot256 committed:
Migrations tested and working

- Added migration for 2.0.0 to 2.x.3 changes.
- Simplified entity validation code.
- Simplified check for when GCKI is removed.
- Added check to renderIcon() for nil target.
- Changed generation of loadedWagonList and loadedWagonMap to procedural, so that car, tarp, and tank are not forgotten if no corresponding vehicles are in the game.
- Fixed bug that on_configuration_change resets all players' tutorial state.

robot256 committed:
Merge branch 'migration_work'

robot256 committed:
Disable wagon swapping events again

- This will ensure the VW error messages stay consistent when replacing wagons intentionally.
- Hopefully preserving the minable property is enough to make other mods happy.

robot256 committed:
Only save entity flags when they are set to false

- Only save entity flags that default to true, if they have been set to false somehow.
- Only restore entity flags if they were explicitly saved as false.

robot256 committed:
Remove potential dependency issue with GCKI

- Only check if GCKI is installed, if VW-GCKI setting is enabled, and if GCKI_data is present to determine if vehicles should be owned/minable or not.

robot256 committed:
Fix last commit

- Only check if GCKI is installed, if VW-GCKI setting is enabled, and if GCKI_data is present to determine if vehicles should be owned/minable or not.

robot256 committed:
Revert "Fix last commit"

This reverts commit 56cab04cae7ef39cf7a7c2914a5d297130978248.

robot256 committed:
Compatibility with UnminableVehicles

- Respect both GCKI and UnminableVehicles when making wagons and stored vehicles minable or unminable.  UnminableVehicles takes priority.
- Apply settings when either mod is installed, uninstalled, or setting changes.

robot256 committed:
Release 2.18.5

- Improved vehicle to wagon mapping reliability.
- Changed wagon_data[x].icon to store the list of render ids.

robot256 committed:
Migrate winches from 1.2.x

- Added migration to convert winch stacks from pairs to singles, searching in players, chests, cars, wagons, and loaded cars.
- Anyone who already migrated their save from 1.2.x gets to keep the extra winches.

robot256 committed:
Update unloadVehicleWagon.lua

- Don't send player_index when restoring equipment grid.

robot256 committed:
Updates

- Fixed icon_mipmaps warning.
- Removed type checking on last_user, since migration takes care of this now.

robot256 committed:
update changelog

robot256 committed:
Create thumbnail_aircraft.png

robot256 committed:
Fix accidental global

robot256 committed:
Release 0.18.6

robot256 committed:
Added support for Spidertron Vehicles

- First version for Factorio 1.0.0
- Add support for Spidertron Vehicles.

robot256 committed:
Update changelog.txt

Remove tabs from changelog

robot256 committed:
Update for Factorio 1.1.0

- Switched to using defines.inventory.spider_xxx now that they fixed the API.
- Switched to using LuaEntity::speed for spidertrons now that they fixed the API.
- Switched to using LuaEntity::selected_gun_index for spidertrons now that they fixed the API.
- Added save and restore of selected_gun_index on both cars and spidertrons.
- Added support saving and restoring the logistic trash inventory on spidertrons.
- Added support for saving and restoring the logistic request slots on spidertrons.
- Added support for saving, but NOT restoring, the auto trash filters on spidertrons.  The API for restoring trash filters is borked at the moment.
- Fixed the "require" file paths in migration scripts, which matters at least when they run on game loading.
- Removed vwtransportercargo from the vehicle blacklist, since it works now.
- Fixed crash in 02-00-03_ui_graphics.lua when drawing icons for loaded mod vehicles when that mod is no longer present.

robot256 committed:
Update to VW 3.1.9

- Changed Winch item to selection-tool so it can be used from the map
- Allowed selecting both vehicle and wagon at the same time to perform loading operation.
- Fixed technology description locale

robot256 committed:
Release 3.1.10

Fix crash when unloading a vehicle without "selected_gun_index" entry.

robot256 committed:
Add option for compatibility loading

Startup setting vehicle-wagon-compatibility-entities changes loaded-vehicle-wagon-XXX loading behavior:
- If the mod for the respective vehicle is present, the sprite and entity are loaded normally.
- Else, if the compatibility setting is checked, a copy of "loaded-vehicle-wagon-tarp" is created with the name of the loaded mod vehicle.

This way, you can load a save after removing another mod without your trains being deleted.  When you try to unload a wagon with a vehicle from a removed mod, it will give an error message and replace with an empty wagon.  The best part is if you don't unload it, you can reinstall the original vehicle mod and then you can unload it.

robot256 committed:
Release 3.1.11

Updated changelog.

robot256 committed:
Added handler for on_entity_cloned

First part of Space Exploration compatibility support.  Now wagon data and icon gets cloned along with the wagon.

robot256 committed:
Error message when using winch in space

- Give the user an error message when selecting a loaded wagon or a vehicle with the winch while in space.
- Uses a slightly hacky way around the Space Exploration remote interface to determine if the surface is a solid or space, or a spaceship (which is also space).

robot256 committed:
Added AAI compatibility

- Treat AAI driver character as not a character.
- Automatically delete AAI driver character before deleting vehicle.
- Use string.find for all the wagon sprite types so that AAI & SE copies get the correct sprites.
- Require AAI version 0.7.9 or above which properly detects unloaded AI vehicles.

robot256 committed:
Fixed crashes when launching SE spaceships

- Catch cases when a VW selection or action is queued while spaceship is launched or anchored (cloned between surfaces).
- Added unit_number to stored data so tables can be properly indexed after the entity is destroyed.

robot256 committed:
Use new SE interface function

- Use the new "get_surface_type" remote interface function.
- Allow loading and unloading spider-vehicles in space.
- Add space-exploration version dependency.
- Update changelog.

robot256 committed:
Release 3.1.12

Forgot the locale update and an extra JSON comma.

robot256 committed:
Release 3.1.13

Fixed that dependencies were mandatory when they should be optional.

robot256 committed:
Bugfixes and updates after SE 0.5.24

- Removed temporary code to delete data for cloned sources.
- Fix that OnEntityCloned did not update the LuaEntity reference to the loaded vehicle wagon entity, and as a result OnConfigurationChanged would purge data for wagons that had been cloned.
- Fix that "Vehicle Lost" message would be shown when script_raised_destroy is called on the source of a cloned entity.
- Fix that dynamic localised names were not handled correctly in some messages.
- Added startup log messages with the vehicle to loaded-wagon entity mapping.
- Removed dependencies, if it crashes just update all the other mods.
- Add support for Lua API Global Variable Viewer.

robot256 committed:
Release 3.1.14

Updated changelog.

robot256 committed:
Create 03-01-14_cloned_entity.lua

Added migration correct bug in 3.1.13.  If you haven't changed any other mods since moving a loaded wagon with a spaceship, it will still be able to unload.

robot256 committed:
Work around GCKI interface error

- GCKI only works with cars, not spider-vehicles.

robot256 committed:
Release 3.1.15

Update changelog

robot256 committed:
Added remote interface to get and set vehicle wagon contents

robot256 committed:
Tested remote interface, 3.1.16

Fixed bugs in interface defintion and crash in unloadVehicleWagon when wagon_data.items is nil.

robot256 committed:
Added Spidertron name support (3.1.17)

Added support for LuaEntity::entity_label, added in Factorio 1.1.9 to read and write the Spidertron custom name field.

robot256 committed:
Updated API

- Commented out log messages.
- Added response to script_raised_destroy::cloned parameter.

robot256 committed:
Added "kill_wagon_data" interface (v3.1.19)

- Added remote interface function `kill_wagon_data(lost_data, unit_number)` to handle the case where another mod deletes a stored vehicle.

dimonft committed:
Update ru.lua

Localization for Russian lang, according to the en.lua template

robot256 committed:
Fixed bug in remote interface

Fixed bug in remote interface function `kill_wagon_data`.

robot256 committed:
Merge pull request #10 from robot256/1.1-support

Merge 1.1 support branch to master

robot256 committed:
Release 3.1.20

- Fixed crash when using "kill_wagon_data" interface function.
- Updated Russian translation. (Thanks to dimonft)

dimonft committed:
Update ru.cfg

Файл локализации актуализирован под современные требования.

dimonft committed:
Fixed a bug in the name of the variable. 

Исправлена ошибка в названии переменной.
Из-за этой ошибки, название Транспортного вагона не отображалось корректно в окне технологий.

EN: Due to this error, the name of the Transport Wagon was not displayed correctly in the technology window.

robot256 committed:
Space Exploration compatibility fixes

robot256 committed:
Updates for GCKI interface, release TBD.

robot256 committed:
New compatibility for Autodrive and GCKI 1.1.3

Contribution from PI-C.

robot256 committed:
Updating minable property handling

- Detect when the new unminable-vehicles mod is installed (though it really doesn't matter because that mod changes prototypes instead of LuaEntity::minable).
- Correctly restore unloaded wagon's minable property after unloading a GCKI vehicle.
- Correctly restore loaded vehicle minable and operable properties when GCKI is uninstalled.

robot256 committed:
Release 3.1.22

- removed debug prints, updated locale
- updated readme and changelog

robot256 committed:
Updated Readme for 3.1.22

robot256 committed:
Added event filters that work properly

Added event filters for all events that support it.
They have to be registered in the OnLoad function so that we can use global tables in the filter lists.

robot256 committed:
Fix crash when marking car for deconstruction

robot256 committed:
Release 4.0.0 for Factorio 2.0

  Features:
    - Updated for Factorio 2.0
    - Made the Winch into a shortcut and custom input.
    - Added Tips and Tricks simulation for vehicle load/unload operations.
    - Added support for colored cars and tanks showing their color when loaded on wagons.
    - Spidertrons can load themselves when instructed to follow an empty vehicle wagon.
  Changes:
    - Removed the "hint" console spam in favor of the Tips and Tricks page.
    - Overhauled vehicle storage to teleport them to a hidden surface instead of deleting the entity.
    - Perfectly retains equipment grids, quality, and spoilage with no overhead.
    - Added event filters and object registration for better performance.
    - Renamed "winch" item to "winch-tool".
    - Prevent unloading vehicles on elevated rails.
    - Most console error messages changed to flying text notifications.

robot256 committed:
Updated publishing script

robot256 committed:
Added check if migration has already been run

- Fixed crash when migrating from previous test builds.

robot256 committed:
Renamed locale file(s) to locale.cfg for use with crowdin.

factorio-mods-helper committed:
Update translations from Crowdin

robot256 committed:
Loading ramps added

- Loading ramps are inserters + dummy chest
- Can be placed next to rail, rotated, flipped
- Finds the rail it can load to, or alternately switches between the rails it can unload from.
- Todo: When new rails are placed, add them to any nearby ramps.

robot256 committed:
Loading ramps added

- Loading ramps are inserters + dummy chest
- Can be placed next to rail, rotated, flipped
- Finds the rail it can load to, or alternately switches between the rails it can unload from.
- Todo: When new rails are placed, add them to any nearby ramps.

robot256 committed:
Loading ramp placement complete

- Placing rails searches for nearby ramps to attach to.
- Uses on_nth_tick to process rails in batches since they are often built in large groups.
- Rotating and flipping ramps works as expected.
- Rotating ramps with multiple valid unloading rails cycles through them before actually rotating.
- Committed the working prototype file finally.

robot256 committed:
Loading ramp placement complete

- Placing rails searches for nearby ramps to attach to.
- Uses on_nth_tick to process rails in batches since they are often built in large groups.
- Rotating and flipping ramps works as expected.
- Rotating ramps with multiple valid unloading rails cycles through them before actually rotating.
- Committed the working prototype file finally.

robot256 committed:
Release 4.0.2: Added option for inventory slots

Features:
  - Added new startup setting to configure the Vehicle Wagon inventory slots.  Can be set to "None", "Full only", or "Full and empty".
  - Re-added the render icons code to show wagon contents when wagon inventory is not used.
  - Changed loading procedure to detect when inserting the dummy item fails, and draw the sprite icon instead.
  - Added migration code in on_configuration_changed to update the icon/dummy item state every time the inventory-slots setting is changed.

Changes:
  - Removed some commented-out compatibility code.

factorio-mods-helper committed:
Update translations from Crowdin

robot256 committed:
Release 4.0.3 - fixed dumb crash

Fixed crash when in the inventory slots setting logic.

robot256 committed:
Release 4.0.4: Created graphics library

robot256 committed:
Added Active Ramp detection

- Detects stopped trains with vehicle wagons using on_train_changed_state
- Stores active ramp-wagon pairs in storage.active_wagons.
- Maintains list when train departs station, is destroyed, wagon is destroyed, or ramp is destroyed.

robot256 committed:
Added Active Ramp detection

- Detects stopped trains with vehicle wagons using on_train_changed_state
- Stores active ramp-wagon pairs in storage.active_wagons.
- Maintains list when train departs station, is destroyed, wagon is destroyed, or ramp is destroyed.

robot256 committed:
Finished autoloading action code

- Fixed pipetting loaded wagons (using placeable-by, and removed on_player_pipette code entirely)
- Finished automatic loading and unloading code.
- Uses random tick retry delay for reading circuit network when ramp is disabled or vehicle is not in loading zone (could be changed eventually)
- Updated all the loading/unloading/selecting code to work with player_index = -1 for script-initiated actions.

robot256 committed:
Release 4.0.5: Fix Pipette bug

- Fixed crash when pipetting loaded wagons by using built-in pipette logic now.

robot256 committed:
Release 4.1.0: Automatic Loading Ramps!

Features:
    - Added Loading Ramps for automatic vehicle loading and unloading.
    - Loading ramps act like inserters with circuit and logistic connection ability.
    - Unloaded vehicles will point in the direction of the loading ramp inserter arrow.
    - Recommended to use with the "inventory slot" option so train schedule can detect when vehicle cargo is loaded or unloaded.
    - Highly experimental, please report issues.

robot256 committed:
Merge branch 'loading_ramps_2.0' of https://github.com/robot256/Vehicle-Wagon-2 into loading_ramps_2.0

factorio-mods-helper committed:
Update translations from Crowdin

robot256 committed:
Fix missing locale

- Rename technology "vehicle-ramps" to "vehicle-loading-ramps"
- Fix missing locale key for loading ramp tech

robot256 committed:
Release 4.1.1 - loading ramp filters

- Change loading ramp re-check delay to fixed 30 ticks instead of random.
- Added checks to respect inserter filter settings, including quality comparisons.
- Commented out debug prints for ramps

robot256 committed:
Release 4.1.2

Fixed crash when placing rails if other things delete them.

robot256 committed:
Release 4.1.3 - RSO compat

robot256 committed:
Release 4.1.4 - Hide surface in remote view

factorio-mods-helper committed:
Update translations from Crowdin

robot256 committed:
Release 4.1.5

- Removed debug prints

factorio-mods-helper committed:
Update translations from Crowdin

robot256 committed:
Fix invalid locale string in loading ramp tooltip

factorio-mods-helper committed:
Update translations from Crowdin

robot256 committed:
Release 4.1.6 - fix AAI compat

  Changes:
    - Updated translations from Crowdin.
    - Use the new "minable_flag" property instead of "minable".
  Bugfixes:
    - Fixed crash when loading AAI "AI" vehicles by doing a better job of determining what item is associate with the vehicle entity.

factorio-mods-helper committed:
Update translations from Crowdin

robot256 committed:
Release 4.1.7 - wagon speed and sim bug

robot256 committed:
Fix some crashes when loading ramps are destroyed.

robot256 committed:
Release 4.1.8

robot256 committed:
Added better icons for loading ramps

robot256 committed:
Script cleanup from Luachecker

- Fixed that maxWeight check was not being re-applied during data-final-fixes.
- Fixed multiple potential crashes in migration script.
- Fixed potential crash when robot unloads invalid wagon.
- Cleaned up many unused and redundant variables.

robot256 committed:
Compat update with Autodrive and GCKI

Changes from PI-C to work with 2.0 versions of Autodrive and GCKI.

robot256 committed:
Release candidate 4.1.9

robot256 committed:
Ramp bug fixes

- Fix error when cloning loading ramps with editor mode.
- Fix error when a ramp is removed or rotated and the ramp's specified loading rail is not in storage.loading_rails for some reason.

robot256 committed:
Fix bugs!

- Found and fixed the copy-pasted error that resulted in inconsistent ramp data tables.
- Removed the checks that were added to detect inconsistent data tables, so it will "fail fast" the next time an inconsistency appears.

robot256 committed:
Release 4.1.9

- Added a migration to fix corrupted ramp data tables in existing save files.
- Removed console spam when unloading with ramps.

robot256 committed:
Fix bug in rail table migration, add logging

robot256 committed:
Rename migration so it runs again

robot256 committed:
Release 4.1.10: fix spurious renderings

- Fixed that automatic loading and unloading was rendering shapes and never deleting them (fixing a potential source of save file bloat/memory leak).
- Added migration script to clean up old rendering objects.

robot256 committed:
More fixes from Pi-C

- Add locale keys that changed in the GCKI/Autodrive code.
- Properly handle the "active" field and data table entry in migrateLoadedWagon.

robot256 committed:
Release 4.1.11

robot256 committed:
Allow loading and unloading with drivers

robot256 committed:
Unsuccessful attempts to work around AAI Prog Vehicles Bug

robot256 committed:
Update entities-compatibility.lua

Compatibility entities only check entity names, not mod names.  This helps maintain continuity when mods are forked.

robot256 committed:
SE 0.7 compatibility

factorio-mods-helper committed:
Update translations from Crowdin

robot256 committed:
Release 4.1.12

Closes #22

robot256 committed:
Release 4.1.13 - Fix non-operable wagons

    - Fixed that vehicle wagons could not be repaired by the player.
    - Fixed that vehicle wagon equipment grids could not be opened.
    - Ensure that quality can't change vehicle wagon inventory size.

robot256 committed:
Fix loading ramps on horizontal rails

Discovered that math2d.bounding_box.contains_point doesn't use bounding box orientation, so for non-vertical wagons this was a problem. Fixed with plain old vector math.

robot256 committed:
Loading players final fixes

- Removed code blocking players from empty wagons.
- Added allow_passengers=false flag to empty wagon.
- Removed code canceling winch when player enters a vehicle.
- Removed code checking for AI character in passenger seat.
- Bumped dependency version to make sure we get the bug fix.

robot256 committed:
Release 4.1.15

Fixed crash with loading ramps on vertical rails.

factorio-mods-helper committed:
Update translations from Crowdin

robot256 committed:
Fixed compatibility with Krastorio2

Advanced tank graphics path updated.
Added color mask layer.

robot256 committed:
Release 4.1.16

- Prevent loading of vehicles and wagons that are marked for deconstruction.

factorio-mods-helper committed:
Update translations from Crowdin
